### PR TITLE
feat: Sync from aztec-packages

### PR DESCRIPTION
**Pull request by AztecBot**
Automated pull of Noir development from [aztec-packages](https://github.com/AztecProtocol/aztec-packages).
BEGIN_COMMIT_OVERRIDE
END_COMMIT_OVERRIDE

### CODE REVIEW

**Review by TomAFrench — APPROVED** [review]

**Comment by TomAFrench** [comment]
Needs some tests restored still

File: test_programs/compile_success_empty/impl_with_where_clause/src/main.nr [hunk 1]
```diff
@@ -0,0 +1,27 @@
+fn main() {
```
Comment (vezenovm) [inline test_programs/compile_success_empty/impl_with_where_clause/src/main.nr:1]: This test was renamed to `trait_impl_with_where_clause` in https://github.com/noir-lang/noir/pull/5320 and can be removed

File: test_programs/compile_success_empty/reexports/src/main.nr [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use reexporting_lib::{FooStruct, MyStruct, lib};
+use dep::reexporting_lib::{FooStruct, MyStruct, lib};
```
Comment (vezenovm) [inline test_programs/compile_success_empty/reexports/src/main.nr:1]: Is this a regression?

Comment (TomAFrench) [reply test_programs/compile_success_empty/reexports/src/main.nr:1]: I don't have strong feelings on this now tbh. `dep::` is more explicit so 🤷

File: test_programs/compile_success_empty/impl_with_where_clause/src/main.nr (hunk 1 above)
Comment (TomAFrench) [reply test_programs/compile_success_empty/impl_with_where_clause/src/main.nr:1]: Ok, can you delete this so we can merge this? I'm done for the day.

Comment (vezenovm) [reply test_programs/compile_success_empty/impl_with_where_clause/src/main.nr:1]: Will do

File: test_programs/compile_success_empty/reexports/src/main.nr (hunk 2 above)
Comment (vezenovm) [reply test_programs/compile_success_empty/reexports/src/main.nr:1]: I won't block for this, but just wanted to note. Are we not looking to eventually error on `dep::`?

File: test_programs/compile_success_empty/impl_with_where_clause/src/main.nr (hunk 1 above)
Comment (TomAFrench) [reply test_programs/compile_success_empty/impl_with_where_clause/src/main.nr]: Thanks!

**Pull request merged**